libs/common/src/types/mongo-db-models/db-model.ts

```diff
@@ -0,0 +1,25 @@
+import { WithId } from 'mongodb';
+import { MongoDbCollectionName } from './mongo-db-collection-name';
+import { Bank } from './banks';
+import { PortalUser } from './users';
+import { TfmUser } from './tfm-users';
+import { TfmTeam } from './tfm-teams';
+import { CronJobLog } from './cron-job-logs';
+import { UtilisationReport } from './utilisation-reports';
+import { UtilisationData } from './utilisation-data';
+
+export type DbModel<TCollectionName extends MongoDbCollectionName> = TCollectionName extends 'banks'
+  ? Bank
+  : TCollectionName extends 'users'
+  ? PortalUser
+  : TCollectionName extends 'tfm-users'
+  ? TfmUser
+  : TCollectionName extends 'tfm-teams'
+  ? TfmTeam
+  : TCollectionName extends 'cron-job-logs'
+  ? CronJobLog
+  : TCollectionName extends 'utilisationReports' // TODO FN-1853 Remove after SQL refactor
+  ? UtilisationReport
+  : TCollectionName extends 'utilisationData'
+  ? UtilisationData
+  : WithId<object>;
```

Comment:
will grab you offline for this too if that's alright!

Reply:
Discussed: the decision has been made to replace `object` with `any` because
1. Using something like `Record<string, unknown>` does fix an issue with properties not being defined on `object`, but causes errors if you don't stringently check the type of the property (feature of `unknown`)
2. Some of the types that `any` is covering here are very complicated to document and will likely never have types created for them